src/chapters/errors-and-debugging/asking-good-questions.rst

```diff
@@ -0,0 +1,42 @@
+Asking Good Questions
+=====================
+
+If employing the strategies outlined in this chapter still don't help you find a bug in your code, you always have other programmers to reach out to.
+
+Whether you are asking a question of your teaching assistant, another student, or a stranger in an online forum, you should first be able to answer the 3 questions discussed in this section.
+
+Not only will these questions help others assist you more directly, but they may just lead you to the answer yourself! 
+
+What is the problem with your code?
+-----------------------------------
+
+You should be able to state in general terms what the error that you are experiencing is. 
+
+**Bad:** "My program is broken."
+
+**Bad:** "I'm getting this error."
+
+**Good:** "There is a ReferenceError on line 23, but it's not clear to me what's causing it."
+
+What have you done to try to address the problem?
+-------------------------------------------------
+
+Another programmer can glean a lot of information by hearing what you have already tried. 
+
+**Bad:** Asking for help immediately.
+
+**Bad:** Using trial and error without any specific direction.
+
+**Good:** "I added user input validation, but am still seeing the problem."
+
+Where have you looked for an answer?
+------------------------------------
+
+**Bad:** "I haven't looked online at all."
+
+**Bad:** "I Googled 'js range error' and didn't see anything."
+
+**Good:** "I Googled 'js range error boolean expression' and found a question on Stack Overflow that seemed relevant. I tried the the recommended solution, but it didn't fix my problem."
+
+
```

Comment:
Suggestion for another heading:  The only dumb question is the one that isn't asked.
Encouragement: 1 - 2 sentences.

(Feel free to reference the quote.  It's mine.)  ;-)

Reply:
I generally agree with this sentiment, but it _is_ a bit at odds with some of the points we are making in this section. One example is, "**Bad:** Asking for help immediately."

Reply:
Ok.